darts/models/regression_model.py

```diff
@@ -13,300 +13,259 @@
 import numpy as np
 import pandas as pd
 
-from abc import ABC, abstractmethod
-from typing import List, Iterable, Union, Any, Callable
 
+from .. import metrics
 from ..timeseries import TimeSeries
-from ..logging import raise_if_not, get_logger, raise_log
+from sklearn.linear_model import LinearRegression
+from .forecasting_model import ExtendedForecastingModel
+from typing import List, Iterable, Union, Any, Callable, Optional
+from ..logging import raise_if, raise_if_not, get_logger, raise_log
 from ..utils import (
     _build_tqdm_iterator,
     _with_sanity_checks,
     _historical_forecasts_general_checks
 )
-from .. import metrics
 
 logger = get_logger(__name__)
 
 
 # TODO: Extend this to a "DynamicRegressiveModel" class, which acts on List[List[TimeSeries]].
 # TODO: The first List[] would contain time-sliding lists of time series, letting the model
 # TODO: be able to learn how to change weights over time. When len() of outer List[] is 0 it's a particular case
-class RegressionModel(ABC):
-    @abstractmethod
-    def __init__(self):
-        """ Regression Model.
-
-            This is the base class for all regression models.
-        """
+class RegressionModel(ExtendedForecastingModel):
+    def __init__(self,
+                 lags: Union[int, list] = None,
+                 lags_exog: Union[int, list, bool] = None,
+                 model=None):
+        """ Regression Model
 
-        # Stores training date information:
-        self.train_features: List[TimeSeries] = None
-        self.train_target: TimeSeries = None
+        Can be used to fit any scikit-learn-like regressor class to predict the target
+        time series with lagged values.
 
-        # state
+        Parameters
+        ----------
+        lags : Union[int, list]
+            Number of lagged target values used to predict the next time step. If an integer is given
+            the last `lags` lags are used (inclusive). Otherwise a list of integers with lags.
+        lags_exog : Union[int, list, bool]
+            Number of lagged exogenous values used to predict the next time step. If an integer is given
+            the last `lags_exog` lags are used (inclusive). Otherwise a list of integers with lags. If False,
+            the value at time `t` is used which might lead to leakage. If True the same lags as for the
```

Comment:
Perhaps give a little more explanation after "which might lead to leakage": when does it (or does it not) lead to leakage.

Reply:
In addition it might be nice to explain how the target and exog time axes relate to each other. I.e. that for fitting they need to have matching time axes, but for predicting `exog` has to start 1 timestamp after the end of the target.

Reply:
I changed the phrasing a little. Do you think it would help to give an example in the documentation of `RegressionModel`.

Reply:
I explained how the input of `exog` should look like in the documentation of `predict()`. It might be quite strict as it is now but hopefully it will be alright for most users. Unfortunately now it is not possible for a user to make predictions in the time frame during training which might be useful for some users to get training errors. Do you think this needs to be fixed?

Reply:
I think we can merge without having this in place, but it would be nice to keep track of it in our roadmap.